meta-resin-common/recipes-containers/hostapp-update/files/hostapp-update

```diff
@@ -0,0 +1,83 @@
+#!/bin/sh
+
+set -eu
+
+local_image=""
+remote_image=""
+reboot=0
+
+while getopts 'f:i:r' flag; do
+	case "${flag}" in
+        f) local_image=$(realpath "${OPTARG}") ;;
+        i) remote_image="${OPTARG}" ;;
+	r) reboot=1 ;;
+        *) error "Unexpected option ${flag}" ;;
+	esac
+done
+
+export DOCKER_HOST="unix:///var/run/docker-host.sock"
+SYSROOT="/mnt/sysroot/inactive"
+
+# Initialize sysroot
+mkdir -p /mnt/sysroot
+for dir in 'dev' 'etc' 'docker' 'hostapps' 'mnt/state' 'proc' 'sbin' 'sys' 'tmp'; do
+	if [ ! -d "$SYSROOT/$dir" ]; then
+		mkdir -p "$SYSROOT/$dir"
+	fi
+done
+if [ ! -f "$SYSROOT/etc/machine-id" ]; then
+	touch "$SYSROOT/etc/machine-id"
+fi
+if [ ! -L "$SYSROOT/sbin/init" ]; then
+	ln -sf /current/bootstrap/init "$SYSROOT/sbin/init"
+fi
+
+# Remove previous hostapp
+for container in $(docker ps --all --quiet); do
+	docker rm --force --volumes "$container"
+done
+for image in $(docker images --all --quiet); do
+	docker rmi --force "$image"
+done
+for bootstrap in "$SYSROOT/hostapps/"*; do
+	[ -e "$bootstrap" ] || break
+	rm -rf "$bootstrap"
```

Comment:
I'm not 100% sure what this loop is doing, perhaps a comment could clear that up?